test/fail_compilation/aliasassign.d

```diff
@@ -0,0 +1,21 @@
+/* TEST_OUTPUT:
+---
+fail_compilation/aliasassign.d(13): Error: `B` must have same parent `Swap!(int, string)` as alias `B`
+fail_compilation/aliasassign.d(14): Error: `A` must have same parent `Swap!(int, string)` as alias `A`
```

Comment:
This error message doesn't make much sense. "B must have same parent as B"... Well it does, doesn't it ?
And changing the outter `B` (L19) doesn't help.

Reply:
If you name all your symbols the same, no, it isn't going to make that much sense. If you have a better idea, let me know.

Reply:
Note that in your example the parameter `T` is a different declaration than the target `T` of the aliasassign, and they have different parents. They have to have the same parent to be accepted.

Reply:
> If you name all your symbols the same, no, it isn't going to make that much sense. If you have a better idea, let me know.

That's why I mentioned changing the outer `B`:
```D
template Swap (alias A, alias B)
{
    alias C = A;
    B = A;
    A = B;
    enum Swap = true;
}

alias OuterA = int;
alias OuterB = string;

static assert(Swap!(OuterA, OuterB));
```

This produces the same error message.

> Note that in your example the parameter `T` is a different declaration than the target `T` of the aliasassign, and they have different parents. They have to have the same parent to be accepted.

How can they be different, there is no other alias declaration with the same name ?
Looking at the code again, did you mean something like:
```
fail_compilation/aliasassign.d(14): Error: alias `A` must be declared inside `Swap!(int, string)`
```
Likewise, the following would be more user-friendly:
```
fail_compilation/aliasassign.d(14): Error: alias parameter `A` cannot be re-assigned
(errorSupplemental) Only aliases declared inside a template can be re-assigned
```

Reply:
done